File: src/quine.cs
using System;

namespace quine
{
	class quine
	{
		static void Main()
		{
			const string data = @"using System;{2}{2}namespace quine{2}{{{2}	class quine{2}	{{{2}		static void Main(){2}		{{{2}			const string data = @{1}{0}{1};{2}			Console.WriteLine(data, data, '{1}', Environment.NewLine);{2}		}}{2}	}}{2}}}";
			Console.WriteLine(data, data, '"', Environment.NewLine);
		}
	}
}

File: src/quine.scala
object quine extends App {
	val quote = 34.toChar
	var newLine = 10.toChar
	val data = """object quine extends App {%2$s	val quote = 34.toChar%2$s	var newLine = 10.toChar%2$s	val data = %1$s%1$s%1$s%3$s%1$s%1$s%1$s%2$s	println(data format (quote, newLine, data))%2$s}"""
	println(data format (quote, newLine, data))
}
